src/libs/metadata_view.c

```diff
@@ -791,6 +798,112 @@ void connect_key_accels(dt_lib_module_t *self)
+  GtkWidget *area = gtk_dialog_get_content_area(GTK_DIALOG(dialog));
+
+  GtkWidget *w = gtk_scrolled_window_new(NULL, NULL);
+  gtk_widget_set_size_request(w, -1, DT_PIXEL_APPLY_DPI(300));
+  gtk_scrolled_window_set_policy(GTK_SCROLLED_WINDOW(w), GTK_POLICY_AUTOMATIC, GTK_POLICY_AUTOMATIC);
+  gtk_box_pack_start(GTK_BOX(area), w, TRUE, TRUE, 0);
+
+  GtkWidget *grid = gtk_grid_new();
+  gtk_container_add(GTK_CONTAINER(w), grid);
+  GtkWidget *label = gtk_label_new(_("metadata"));
+  gtk_grid_attach(GTK_GRID(grid), label, 0, 0, 1, 1);
+  label = gtk_label_new(_("hidden"));
+  gtk_grid_attach(GTK_GRID(grid), label, 1, 0, 1, 1);
+  gtk_widget_set_tooltip_text(GTK_WIDGET(label),
+            _("tick if the corresponding metadata is of no interest for you"));
+
+  char *pref = dt_conf_get_string("plugins/lighttable/metadata_view/pref_hidden");
+  GList *prefs = dt_util_str_to_glist(",", pref);
+  g_free(pref);
+  GtkWidget *hidden[md_size];
+  gboolean hide[md_size];
+  for(int i = 0; i < md_size; i++)
+  {
+    if(i >= md_xmp_metadata && i < md_xmp_metadata + DT_METADATA_NUMBER &&
+       !_is_metadata_visible(i - md_xmp_metadata))
+      continue;
+    label = dt_ui_label_new(_md_labels[i]);
+    gtk_widget_set_hexpand(label, TRUE);
+    gtk_grid_attach(GTK_GRID(grid), label, 0, i+1, 1, 1);
+
+    hidden[i] = gtk_check_button_new();
+    gtk_widget_set_halign(hidden[i], GTK_ALIGN_CENTER);
+    hide[i] = g_list_find_custom(prefs, _labels[i], (GCompareFunc)g_strcmp0) != NULL;
+    gtk_toggle_button_set_active(GTK_TOGGLE_BUTTON(hidden[i]), hide[i]);
+    gtk_grid_attach(GTK_GRID(grid), hidden[i], 1, i+1, 1, 1);
+  }
+
+#ifdef GDK_WINDOWING_QUARTZ
+  dt_osx_disallow_fullscreen(dialog);
+#endif
+  gtk_widget_show_all(dialog);
+
+  if (gtk_dialog_run(GTK_DIALOG(dialog)) == GTK_RESPONSE_YES)
+  {
+    for(int i = 0; i < md_size; i++)
+    {
+      if(i >= md_xmp_metadata && i < md_xmp_metadata + DT_METADATA_NUMBER &&
+         !_is_metadata_visible(i - md_xmp_metadata))
+        continue;
+      const gboolean hide_new = gtk_toggle_button_get_active(GTK_TOGGLE_BUTTON(hidden[i]));
+      if(hide_new && !hide[i])
+      {
+        char *needle = g_strdup(_labels[i]);
+        prefs = g_list_prepend(prefs, needle);
+      }
+      else if(!hide_new && hide[i])
+      {
+        GList *prefs2 = g_list_find_custom(prefs, _labels[i], (GCompareFunc)g_strcmp0);
+        prefs = g_list_remove_link (prefs, prefs2);
+        g_free(prefs2->data);
+        g_list_free(prefs2);
+      }
+    }
+    pref = dt_util_glist_to_str(",", prefs);
+    dt_conf_set_string("plugins/lighttable/metadata_view/pref_hidden", pref);
+    g_free(pref);
+    _update_layout(self);
+  }
+  g_list_free_full(prefs, g_free);
+  gtk_widget_destroy(dialog);
+}
+
+void set_preferences(void *menu, dt_lib_module_t *self)
+{
+  GtkWidget *mi = gtk_menu_item_new_with_label(_("preferences..."));
+  g_signal_connect(G_OBJECT(mi), "activate", G_CALLBACK(_menuitem_preferences), self);
```

Comment:
The value of self received in _menuitem_preferences doesn't match the sent value ...

Reply:
this is because you don't have the right args in _menuitem_preferences I don't recall the exact signature of the activate callback for menuitems, but first arg is always the widget... (I've faced same problem in modulegroups because I've forgot the event at the second place :))

Reply:
Thanks ! my _menuitem_preferences is incorrect indeed !